pkg/utils/informer/informer.go

```diff
@@ -59,6 +63,19 @@ func (i *Informer[T, L]) Sync(ctx context.Context, opt Option, events chan<- Eve
 		events <- Event[T]{Type: Sync, Object: obj.(T)}
 		return nil
 	}
+
+	if opt.EnableStreamWatch {
+		if watchListOptions, hasWatchListOptionsPrepared, watchListOptionsErr := watchlist.PrepareWatchListOptionsFromListOptions(opt.toListOptions()); watchListOptionsErr != nil {
+			log.FromContext(ctx).Error("Failed preparing watchlist options, falling back to the standard LIST semantics", watchListOptionsErr)
+		} else if hasWatchListOptionsPrepared {
+			watcher, err := i.WatchFunc(ctx, watchListOptions)
+			if err != nil {
+				return err
+			}
+			return handleWatchList(ctx, watcher)
```

Comment:
Events needs to be passed to the channel.

```suggestion
			return handleWatchList(ctx, watcher, events)
```